### PR TITLE
init: init with values

### PR DESCRIPTION
**Pull request by jack-lvsjack**
**NOTE**

1. The protocol is usually listed within 24 hours of merging the PR
2. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
3. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
4. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
5. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by jack-lvsjack** [comment]
This implementation is basically the same as the implementation of stalay, when calling the test, this error will appear
```
❯ node test.js projects/sio2/index.js

 ------ ERROR ------ 


Error: invalid BigNumber value (argument="value", value="78091.298362530687228705", code=INVALID_ARGUMENT, version=bignumber/5.6.2)
Truncated error stack:
    at Function.BigNumber.from (/Users/Jack/Developer/sio2_defiliama/node_modules/@ethersproject/bignumber/lib/bignumber.js:239:23)
    at /Users/Jack/Developer/sio2_defiliama/test.js:331:16
    at Array.forEach (<anonymous>)
    at fixBalances (/Users/Jack/Developer/sio2_defiliama/test.js:325:28)
    at computeTVL (/Users/Jack/Developer/sio2_defiliama/test.js:337:3)
    at getTvl (/Users/Jack/Developer/sio2_defiliama/test.js:60:30)
```

same with starlay:
```
❯ node test.js projects/starlay/index.js

 ------ ERROR ------ 


Error: invalid BigNumber value (argument="value", value="61777589.239082870987730374", code=INVALID_ARGUMENT, version=bignumber/5.6.2)
Truncated error stack:
    at Function.BigNumber.from (/Users/Jack/Developer/sio2_defiliama/node_modules/@ethersproject/bignumber/lib/bignumber.js:239:23)
    at /Users/Jack/Developer/sio2_defiliama/test.js:331:16
    at Array.forEach (<anonymous>)
    at fixBalances (/Users/Jack/Developer/sio2_defiliama/test.js:325:28)
    at computeTVL (/Users/Jack/Developer/sio2_defiliama/test.js:337:3)
    at getTvl (/Users/Jack/Developer/sio2_defiliama/test.js:60:30)
```

please check it, and please also confirm that all the token address is correct.

File: projects/sio2/constanrs.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const DEFAULT_DECIMALS = 18
```
Comment (joseph-w3) [inline projects/sio2/constanrs.js]: Let's change this filename to constants.js

File: projects/sio2/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const { getReserves, getSio2Tvl } = require("./sio2");
+const BigNumber = require("bignumber.js");
+const { getBorrowed } = require("../helper/aave");
+const { TOKENS, SIO2_TOKEN, DOT_DECIMALS, DEFAULT_DECIMALS, LAY_ADDRESS } = require("./constanrs");
```
Comment (joseph-w3) [inline projects/sio2/index.js]: SIO2_ADDRESS

File: projects/sio2/sio2.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
+const sdk = require("@defillama/sdk");
+const abi = require("./abi.json");
+const { default: BigNumber } = require("bignumber.js");
+const addressesProviderRegistry = "0x9D8bB85b1c728f69672923dD4A0209EC8b75EFda"; // from https://github.com/SiO2-Finance/sio2-protocol/blob/main/deployed-contracts.json
```
Comment (joseph-w3) [inline projects/sio2/sio2.js]: Let's change the comment to refer to our documentation: https://sio2-finance.gitbook.io/en/development/contract-addresses

**Comment by jack-lvsjack** [comment]
I tested with built-in method found everything was all right. 

image

## However 
I found that other projects (Kayla) also show the same Error like Starlay: `invalid BigNumber value`, so I guess this error only happen when testing.

I have modified the file to be the same as the other projects, because the other project is running normally. we will see the result.  @joseph-web3

**Pull request merged**